From a08fa8d12df33c67f00b5b3325e8c316022b3082 Mon Sep 17 00:00:00 2001
From: Dudemanguy <random342@airmail.cc>
Date: Sun, 4 Oct 2026 11:56:46 -0500
Subject: [PATCH 2/2] rdp-smartcard: guard logind usage

These functions require libsystemd so the build will fail if systemd is
disabled.
---
 src/grd-rdp-smartcard.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/grd-rdp-smartcard.c b/src/grd-rdp-smartcard.c
index 15e530a..f85f022 100644
--- a/src/grd-rdp-smartcard.c
+++ b/src/grd-rdp-smartcard.c
@@ -1936,6 +1936,7 @@ get_session_id (void)
   if (session_id && session_id[0] != '\0')
     return g_strdup (session_id);
 
+#if HAVE_LIBSYSTEMD
   logind_session_id = grd_get_session_id_from_pid (getpid ());
   if (logind_session_id)
     return logind_session_id;
@@ -1943,6 +1944,7 @@ get_session_id (void)
   logind_session_id = grd_get_session_id_from_uid (getuid ());
   if (!logind_session_id)
     g_warning ("[RDP.SMARTCARD] Failed to get session ID");
+#endif
 
   return logind_session_id;
 }
-- 
2.56.0

